src/installer/destinations.py

```diff
@@ -93,6 +105,8 @@ def __init__(
         interpreter: str,
         script_kind: "LauncherKind",
         hash_algorithm: str = "sha256",
+        bytecode_optimization_levels: Collection[int] = (0, 1),
```

Comment:
-1 means it depends on the running interpreter, if I've followed the Python docs correctly? I think for the Python API, we want to avoid any behaviour that implicitly depends on the interpreter running it. Then again, the bytecode generated is already specific to the minor version of Python we're running on, and I don't think there's any good way to generate bytecode for a different version of Python.

Maybe the default for the Python API should be `()`, i.e. no bytecode compilation by default?

Reply:
🤷🏽 

Let's do either `()` or `(0, 1, 2)`. I'll let you pick. :)

Reply:
I've gone for `()` as the default. At the Python API level, I think people should have to opt in to anything that's specific to the Python version running `installer`.

Reply:
Sounds good!

Reply:
If the paths to install to are specific to the current interpreter, then it seems reasonable for the bytecode to also be.

If the paths to install are specific to a defined `--interpreter` then that can be used to run `-m compileall`.